### PR TITLE
chore: get primary node info from redis sentinel instead KB_LEADER

### PR DESCRIPTION
**Pull request by Y-Rookie**

### CODE REVIEW

File: addons/redis/scripts/redis-start.sh [hunk 1]
```diff
@@ -140,6 +130,93 @@ init_or_get_primary_node() {
...
+  fi
+
+  old_ifs="$IFS"
+  IFS=','
+  set -f
+  read -ra sentinel_pod_list <<< "${SENTINEL_POD_NAME_LIST}"
+  set +f
+  IFS="$old_ifs"
+  declare -A master_count_map
+  local default_redis_primary_host=""
```
Comment (weicao) [inline addons/redis/scripts/redis-start.sh:161]: default_redis_primary_host -> first_redis_primary_host
default_redis_primary_port -> first_redis_primary_port

would be better

File: addons/redis/scripts/redis-start.sh [hunk 2]
```diff
@@ -140,6 +130,93 @@ init_or_get_primary_node() {
...
+      echo "sentinel:$sentinel_pod_fqdn has no master info, skip this sentinel."
+      continue
+    fi
+
+    # increment the count of this master in the map
+    host_port_key="${REDIS_SENTINEL_INFO[0]}:${REDIS_SENTINEL_INFO[1]}"
+    master_count_map[$host_port_key]=$(( ${master_count_map[$host_port_key]} + 1 ))
+
+    # track the primary host and port from the first sentinel
+    if [[ -z "$primary_host" ]] && [[ -z "$primary_port" ]]; then
```
Comment (weicao) [inline addons/redis/scripts/redis-start.sh:182]: do you want to express the following logic ?

if [[ -z "$default_redis_primary_host" ]] && [[ -z "$default_redis_primary_port" ]]; then

Comment (Y-Rookie) [reply addons/redis/scripts/redis-start.sh]: > do you want to express the following logic ?
> 
> if [[ -z "$default_redis_primary_host" ]] && [[ -z "$default_redis_primary_port" ]]; then

yes,  it has been fixed

File: addons/redis/scripts/redis-start.sh (hunk 1 above)
Comment (Y-Rookie) [reply addons/redis/scripts/redis-start.sh]: > default_redis_primary_host -> first_redis_primary_host default_redis_primary_port -> first_redis_primary_port
> 
> would be better

fixed

**Review by weicao — APPROVED** [review]

**Review by ldming — APPROVED** [review]

**Pull request merged**